OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs

```diff
@@ -34,7 +34,7 @@ public ProductionProperties(ScriptContext context, Actor self)
 
 		[ScriptActorPropertyActivity]
 		[Desc("Build a unit, ignoring the production queue. The activity will wait if the exit is blocked.",
-			"If productionType is nil or unavailable, then an exit will be selected based on Buildable info.")]
+			"If productionType is nil or unavailable, then an exit will be selected ignoring production types.")]
```

Comment:
Can we please keep the original Desc here and change the behaviour to match it?
Something like this should work, i think:
```diff
diff --git a/OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs b/OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs
index eb02f93b78..c91f4c34b7 100644
--- a/OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs
+++ b/OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs
@@ -41,12 +41,14 @@ public void Produce(string actorType, string factionVariant = null, string produ
                        if (!Self.World.Map.Rules.Actors.TryGetValue(actorType, out actorInfo))
                                throw new LuaException("Unknown actor type '{0}'".F(actorType));
 
+                       var bi = actorInfo.TraitInfo<BuildableInfo>();
                        Self.QueueActivity(new WaitFor(() =>
                        {
                                // Go through all available traits and see which one successfully produces
                                foreach (var p in productionTraits)
                                {
-                                       if (!string.IsNullOrEmpty(productionType) && !p.Info.Produces.Contains(productionType))
+                                       var type = productionType ?? bi.BuildAtProductionType;
+                                       if (!string.IsNullOrEmpty(type) && !p.Info.Produces.Contains(type))
                                                continue;
 
                                        var inits = new TypeDictionary
@@ -55,7 +57,7 @@ public void Produce(string actorType, string factionVariant = null, string produ
                                                new FactionInit(factionVariant ?? BuildableInfo.GetInitialFaction(actorInfo, p.Faction))
                                        };
 
-                                       if (p.Produce(Self, actorInfo, productionType, inits))
+                                       if (p.Produce(Self, actorInfo, type, inits))
                                                return true;
                                }
 
```

Otherwise we'll be forcing map authors to define redundant (and fragile) information if they have structures that can produce multiple actor types.